### ISSUE TITLE
On Windows the icon stays visible after exiting the app until you move your mouse over it

### ISSUE THREAD

**Issue by alinnert** [zaaack/node-systray#1]
I'm using Windows 10 (1709). I wrote a simple test script and everything works fine so far. But when I exit the app the icon in the systray doesn't get removed. It stays there until I hover the mouse pointer over it. I know this behavior from apps that crash. Windows doesn't seem to recognize that the app isn't running anymore.

I've tried to exit the app the following ways:

~~~js
// v1
systray.kill()
// v2
process.exit(0)
// v3
systray.kill()
process.exit(0)
~~~

None of them makes the icon disappear after the app stops running.

**Comment by zaaack** [comment]
Sorry I haven't tested in windows yet, maybe I should quit in go side, I'll look into this as soon as possible, thanks for your report!

**Comment by zaaack** [comment]
@alinnert Just published a new version (systray@1.0.2), should fixed your issue.

**Comment by zaaack** [comment]
BTW, if you want to implement something on systray, you might also want to check this out: https://github.com/zaaack/aria2c-node-gui (A highly functional demo project to shows how to integrate with opn, node-notifier, and also include pack script by node-packer to create a single executable file).

**Comment by alinnert** [comment]
@zaaack Thank you, I'll try it again this evening.

And thanks, I'll check it out. The funny thing is, I'm already using `opn`. `node-notifier` seems useful, too.

But thanks for mentioning `node-packer`. Currently, I'm using a `.cmd` file and was looking for a way to convert that to a `.exe`. `node-packer` might be worth a try.

**Comment by alinnert** [comment]
@zaaack unfortunately it did not fix the issue. I've looked up the `node_modules/systray` folder. The version got updated to `1.0.2`. Also when I start `tray_windows(_release).exe` on its own (just out of curiousity) it behaves the same way.

**Comment by zaaack** [comment]
Can you run tray_windows.exe in cmd.exe directly, and copy test/menu.json's content to the command prompt, then press enter,and find the tray icon, and press ctrl+c to see if the icon still there?

**Comment by zaaack** [comment]
Can you run tray_windows.exe in cmd.exe directly, and copy test/menu.json's content to the command prompt, then press enter,and find the tray icon, and press ctrl+c to see if the icon still there?

**Comment by zaaack** [comment]
I tried https://github.com/zaaack/aria2c-node-gui  in a win10 machine, and it works well. I think I know why, the systray process is killed without cleaning after nodejs process exit. You should waiting the systray process quit then exit  nodejs process. 

You can find some example code here: https://github.com/zaaack/aria2c-node-gui/blob/master/src/main.ts#L92-L99

**Comment by zaaack** [comment]
Update `systray.kill()` method, now it would automatic quit current nodejs process with less-buggy way, unless you pass [a false param](https://zaaack.github.io/node-systray/classes/_index_.systray.html#kill).

**Comment by alinnert** [comment]
No, it's still not working for me. I've noticed it does work if I call `tray_windows.exe` via `cmd`. It also works if I open the exe with a double click and close the opening console window via `ctrl` + `c`. But it doesn't if a close this window with the "X" in the title bar. But I think that's just how the console works.

I've removed the `process.exit()`, tried `systray.kill()` with `true`, `false` and no parameter. I've also tried your workaround with the current version. But since the exe is working the problem should be on the JS side, right?

**Comment by zaaack** [comment]
If you close the console by close button, the systray process was killed without cleaning,  it need time to notify windows that it's quit, and this is how the trick works.

Are you quit by tray menu?  Actually i tried https://github.com/zaaack/aria2c-node-gui , and quiting by tray menu works here, can you tray this demo

**Comment by alinnert** [comment]
Well, I tried to try the demo project, but it fails at `npm start`. It says that `bin` is an illegal option in `del ./bin/**` (from the `clean` task). But anyway, I'm using `systray.kill()` from inside `systray.onClick()`, just like the demo project. Also the `setTimeout` you mentioned didn't help.
Now I've also tried `systray.kill()` right after initializing it (`new SysTray()`). With a `setTimeout` it's the same behavior. But without `setTimeout` the icon doesn't even show up. Maybe the systray process isn't able to finish loading in that case?

**Comment by zaaack** [comment]
You are right, could you post your code here? Or a minimal project would be best.

**Comment by alinnert** [comment]
[project-hub-src.zip](https://github.com/zaaack/node-systray/files/1411849/project-hub-src.zip)

I've created a ZIP with my current project. The only thing it does is listing all directories inside `C:\projects` (via ajax).

**Comment by zaaack** [comment]
I have reproduced this bug with your code, and my code, too(Strange that I didn't notice it before). I'm not sure I can fix this, but the upstream project having some updates, I'll give a try after I finish my current work (just need several days).

**Comment by genesy** [comment]
This is still an issue :(

**Comment by felixhao28** [comment]
@genesy and generally everyone on this thread, I forked this repo and fixed this issue and [published it as `systray2` on npm](https://www.npmjs.com/package/systray2). I will continue to maintain systray project as [systray2](https://github.com/felixhao28/node-systray) since this project is pretty much abandoned.
